### PR TITLE
Improve flight sql examples

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Implement `get_flight_info_tables` and remove unimplemented methods.

## Are these changes tested?
image

### CODE REVIEW

File: datafusion-examples/examples/flight/flight_sql_server.rs [hunk 1]
```diff
@@ -246,27 +276,6 @@ impl FlightSqlService for FlightSqlServiceImpl {
         Ok(Response::new(Box::pin(stream)))
     }
 
-    async fn get_flight_info_statement(
```
Comment (alamb) [inline datafusion-examples/examples/flight/flight_sql_server.rs:249]: removing this code means it falls back to the default implementation (which also returns Not implemented)

Seems like a nice change to me 👍

File: datafusion-examples/examples/flight/flight_sql_server.rs [hunk 2]
```diff
@@ -337,234 +346,52 @@ impl FlightSqlService for FlightSqlServiceImpl {
...
-    async fn get_flight_info_schemas(
-        &self,
-        _query: CommandGetDbSchemas,
-        _request: Request<FlightDescriptor>,
-    ) -> Result<Response<FlightInfo>, Status> {
-        info!("get_flight_info_schemas");
-        Err(Status::unimplemented("Implement get_flight_info_schemas"))
-    }
-
     async fn get_flight_info_tables(
```
Comment (alamb) [inline datafusion-examples/examples/flight/flight_sql_server.rs:349]: 👍

File: datafusion-examples/examples/flight/flight_sql_server.rs [hunk 3]
```diff
@@ -337,234 +346,52 @@ impl FlightSqlService for FlightSqlServiceImpl {
...
-        _query: CommandGetXdbcTypeInfo,
-        _request: Request<Ticket>,
-    ) -> Result<Response<<Self as FlightService>::DoGetStream>, Status> {
-        info!("do_get_xdbc_type_info");
-        Err(Status::unimplemented("Implement do_get_xdbc_type_info"))
-    }
+        let fetch = FetchResults { handle: uuid };
+        let buf = fetch.as_any().encode_to_vec().into();
+        let ticket = Ticket { ticket: buf };
+        let endpoint = FlightEndpoint {
```
Comment (alamb) [inline datafusion-examples/examples/flight/flight_sql_server.rs:365]: Nice! I think you can make this code simpler using `FlightInfo::new()` as shown on the examples in https://docs.rs/arrow-flight/latest/arrow_flight/struct.FlightInfo.html

```rust
// Create a new FlightInfo
let flight_info = FlightInfo::new()
  // Encode the Arrow schema
  .try_with_schema(&schema))
  .expect("encoding failed")
  .with_endpoint(
     FlightEndpoint::new()
       .with_ticket(ticket)
     )
   )
  .with_descriptor(FlightDescriptor::new_cmd(Default::default()));
```

**Review by alamb — APPROVED** [review]
Thank you @lewiszlw  -- this looks really nice to me

File: datafusion-examples/examples/flight/flight_sql_server.rs (hunk 3 above)
Comment (lewiszlw) [reply datafusion-examples/examples/flight/flight_sql_server.rs]: Thanks for the suggestion!
`.with_descriptor(FlightDescriptor::new_cmd(Default::default()))` can not be compiled due to 
```
cannot infer type for type parameter `impl Into<Bytes>` declared on the associated function `new_cmd`
```
So I changed to 
```
        let info = FlightInfo::new()
            // Encode the Arrow schema
            .try_with_schema(&schema)
            .expect("encoding failed")
            .with_endpoint(FlightEndpoint::new().with_ticket(ticket))
            .with_descriptor(FlightDescriptor {
                r#type: DescriptorType::Cmd.into(),
                cmd: Default::default(),
                path: vec![],
            });
```

File: datafusion-examples/examples/flight/flight_sql_server.rs [hunk 4]
```diff
@@ -304,267 +313,50 @@ impl FlightSqlService for FlightSqlServiceImpl {
...
-    async fn do_put_prepared_statement_query(
-        &self,
-        _query: CommandPreparedStatementQuery,
-        _request: Request<PeekableFlightDataStream>,
-    ) -> Result<Response<<Self as FlightService>::DoPutStream>, Status> {
-        info!("do_put_prepared_statement_query");
-        Err(Status::unimplemented(
-            "Implement do_put_prepared_statement_query",
-        ))
+        let info = FlightInfo::new()
```
Comment (alamb) [inline datafusion-examples/examples/flight/flight_sql_server.rs:348]: 😍

**Review by alamb — APPROVED** [review]
Thanks agian @lewiszlw

**Pull request merged**